ext/compressors/zstd/zstd_compress.c

```diff
@@ -98,15 +183,24 @@ zstd_compress(WT_COMPRESSOR *compressor, WT_SESSION *session, uint8_t *src, size
   uint8_t *dst, size_t dst_len, size_t *result_lenp, int *compression_failed)
 {
     ZSTD_COMPRESSOR *zcompressor;
+    ZSTD_CONTEXT *context = NULL;
     size_t zstd_ret;
     uint64_t zstd_len;
 
     zcompressor = (ZSTD_COMPRESSOR *)compressor;
 
+    zstd_get_context(zcompressor, session, CONTEXT_TYPE_COMPRESS, &context);
+
     /* Compress, starting past the prefix bytes. */
-    zstd_ret = ZSTD_compress(
-      dst + ZSTD_PREFIX, dst_len - ZSTD_PREFIX, src, src_len, zcompressor->compression_level);
+    if (context != NULL) {
+        zstd_ret = ZSTD_compressCCtx((ZSTD_CCtx *)context->ctx, dst + ZSTD_PREFIX,
+          dst_len - ZSTD_PREFIX, src, src_len, zcompressor->compression_level);
+    } else {
+        zstd_ret = ZSTD_compress(
+          dst + ZSTD_PREFIX, dst_len - ZSTD_PREFIX, src, src_len, zcompressor->compression_level);
+    }
 
+    zstd_release_context(zcompressor, session, CONTEXT_TYPE_COMPRESS, context);
```

Comment:
The context is taken and it is released when the compression is finished. Usually, the context is useful when it is reused for multiple compressions? I may be missing something here to understand the advantage of the context used here.

Reply:
The only advantage of this context management is to avoid repeated allocation/free.
If we want to take advantage of using same context across multiple compressions, then we need to associate the context with session. 1. In extension we don't depend on wiredtiger sessions. 2. We use compression at block level, we don't use streaming compression in which cases associating a context to session is more advantageous.  Considering 1 & 2 I choose to use context with a single compression itself.